### DISCUSSION TITLE
How to run a C++ code in Helix

### DISCUSSION THREAD

**Discussion by Abdullah-Mahthel** [helix-editor/helix#7444] [Q&A]
Hi there ,

im using Helix for couple weeks and l like it a lot but 

How can l run C++ code on Helix? Note"im using Windows :/"

any ideas? suggestions ,step by step l know im noob on this kind of editors :)

**Comment by the-mikedavis** [comment]
Building and running projects is not supported by Helix. That should be done in a separate terminal pane/split.

**Reply by Abdullah-Mahthel** [reply]
So l have to use clangd to compile and run c/c++ code ?

**Reply by gabydd** [reply]
yep, though helix has some features you see in lot's of IDEs, it is still an editor that tries to be language agnostic and use well established protocols like lsp and tree-sitter in the core. So for compiling code in the editor that will probably have to wait on plugins or https://github.com/helix-editor/helix/issues/2934

**Comment by quantonganh** [comment]
https://github.com/helix-editor/helix/issues/1976#issuecomment-1645104621
